### PR TITLE
Modify examples to read/write boolean attribute

### PR DESCRIPTION
**Pull request by sergioherrera7**
Run the test Server that sets up a boolean value node:

`build/examples/TestServer/TestServer`

Read/write from the node:

`build/examples/SubscriptionTest/SubscriptionTest`

### CODE REVIEW

File: examples/SubscriptionTest/main.cpp [hunk 1]
```diff
@@ -33,13 +33,65 @@ int main()
...
+                        }
+                    }
+                    else {
+                        cout << " Could not readValueAttribute";
+                    }
+
+                    /////
+                    static bool BOOLvalue = true;
+                    BOOLvalue             = !BOOLvalue;
+                    Open62541::Variant data(BOOLvalue);
```
Comment (sergioherrera7) [inline examples/SubscriptionTest/main.cpp:62]: this always creates a Open62541::Variant that reads "true" when called with .toString()

strange that when written it does toggle the value on the server:

**Comment by sergioherrera7** [comment]
**Do not merge!!**

This is an experimental change to show the bug regarding the boolean attribute read/write

File: examples/SubscriptionTest/main.cpp [hunk 2]
```diff
@@ -33,13 +33,65 @@ int main()
...
-                         << Open62541::dataValueToString(v) << endl;
+                auto f = [&client,
+                          idx](Open62541::ClientSubscription& c, Open62541::MonitoredItem* m, UA_DataValue* v) {
+                    cout << "Data Change SubId " << c.id() << " Monitor Item Id " << m->id() << " Value "
+                         << v->value.type->typeName << " " << Open62541::dataValueToString(v) << endl;
+
+                    cout << endl;
+
+                    Open62541::NodeId nodeNumber(idx, "BOOLEAN_Value");
+                    Open62541::Variant nodeValue;
```
Comment (axelperschmann) [inline examples/SubscriptionTest/main.cpp:46]: I feel like the `Open62541::Variant` must be initialized with a type here!

Something like this. Apparently after reading the value, it does not know it has read a bool?!?!
```
Open62541::Variant<UA_TYPES_BOOLEAN>
```

Comment (sergioherrera7) [reply examples/SubscriptionTest/main.cpp:46]: I also had my suspicions with this one, 
but the abstract Variant constructor does work with all types and initializes the internal data

File: src/open62541objects.cpp [hunk 3]
```diff
@@ -94,7 +94,7 @@ std::string Open62541::variantToString(UA_Variant& v)
             ^^^^^^^
         */
         case UA_DATATYPEKIND_BOOLEAN: {
-            ret = ((UA_Boolean*)(v.data)) ? "true" : "false";
+            ret = (*(UA_Boolean*)(v.data)) ? "true" : "false";
```
Comment (sergioherrera7) [inline src/open62541objects.cpp:97]: Bug found!

File: examples/SubscriptionTest/main.cpp (hunk 1 above)
Comment (sergioherrera7) [reply examples/SubscriptionTest/main.cpp:62]: Fix:
https://github.com/psiori/open62541Cpp/pull/2#discussion_r1151421795

**Comment by ahoereth** [comment]
merge this?

**Comment by sergioherrera7** [comment]
The examples are dirty, I will merge just the bugfix

**Comment by sergioherrera7** [comment]
Already there: 4aa6db46407ee87aea6733c2ed4e65bfed42ab06

**Pull request closed without merging**